shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionActionDispatcher.kt

```diff
@@ -113,6 +117,14 @@ class StepCompletionActionDispatcher(
                     onNewMessage(Message.CheckTopicCompletionStatus.Uncompleted)
                 }
             }
+            is Action.UpdateProblemsLimit ->
+                if (freemiumInteractor.isFreemiumEnabled().getOrDefault(false)) {
+                    currentSubscriptionStateRepository.getState().getOrNull()?.let {
+                        currentSubscriptionStateRepository.updateState(
+                            it.copy(stepsLimitLeft = it.stepsLimitLeft?.dec())
+                        )
+                    }
+                }
```

Comment:
What do u think about moving this logic into `FreemiumInteractor.onStepSolved()`?

Reply:
Ideally, we can add subscription to `solvedStepsSharedFlow` into `FreemiumInteractor`, to let it update itself. 
In that case `FreemiumInteractor` should be a singleton object.